lib/cli/index.js

```diff
@@ -0,0 +1,511 @@
+var _ = require('lodash'),
+    argparse = require('argparse'),
+    fs = require('fs'),
+    async = require('async'),
+    parseJson = require('parse-json'),
+    request = require('request'),
+
+    /**
+     * Separates reporter specific arguments from the rest.
+     *
+     * @todo Figure out a better way! (Or submit a PR to argparse).
+     * @param allArgs
+     */
+    separateReporterArgs = (allArgs) => {
+        var reporterArgs = [],
+            args = [],
+
+            i;
+
+        // Separate the reporter arguments from the rest
+        for (i = 0; i < allArgs.length; i++) {
+            if (_.startsWith(allArgs[i], '--reporter-')) {
+                reporterArgs.push(allArgs[i]);
+
+                // Also push the next parameter if it's not an option.
+                allArgs[i + 1] && !_.startsWith(allArgs[i + 1], '-') && reporterArgs.push(allArgs[++i]);
+            }
+            else {
+                args.push(allArgs[i]);
+            }
+        }
+        return {
+            reporter: reporterArgs,
+            argv: args
+        };
+    },
+
+    /**
+     * Loads JSON data from the given location.
+     *
+     * @param {string} location - Can be an HTTP URL or a local file path.
+     * @param {object=} options
+     * @param {object} options.apikey - Postman's cloud API Key (if the resource is being loaded from Postman Cloud)
+     * @param callback
+     * @returns {Promise}
+     */
+    fetch = function (location, options, callback) {
+        !callback && _.isFunction(options) && (callback = options, options = {});
+        return (/^https?:\/\/.*/).test(location) ?
+            // Load from URL
+            request.get({ url: location, json: true }, (err, response, body) => {
+                if (err) {
+                    return callback(err);
+                }
+
+                try {
+                    _.isString(body) && (body = parseJson(body));
+                }
+                catch (e) {
+                    return callback(new Error('Given URL did not provide a valid JSON response: ', location));
+                }
+
+                return callback(null, body);
+            }) :
+            fs.readFile(location, function (err, value) {
+                if (err) {
+                    return callback(err);
+                }
+
+                try {
+                    value = parseJson(value.toString());
+                }
+                catch (e) {
+                    callback(new Error('Given file does not contain valid JSON data: ', location));
+                }
+
+                return callback(null, value);
+            });
+    },
+
+    /**
+     * Creates a parser for the older CLI arguments. (Without subcommands).
+     *
+     * @param program
+     * @returns {*}
+     */
+    execute = (program) => {
```

Comment:
I've kept _all_ the legacy params as they are, since they're not shown anywhere as of now (not even in the help text).